File: DSA/Ayush_Mishra_590014844/Day_57_Question_2.c
#include <stdio.h>

int isMaxHeap(int arr[], int n) {
    for (int i = 0; i <= (n - 2)/2; i++) {
        if (arr[i] < arr[2*i + 1])
            return 0;
        if (2*i + 2 < n && arr[i] < arr[2*i + 2])
            return 0;
    }
    return 1;
}

int main() {
    int n;
    printf("Enter size of array: ");
    scanf("%d", &n);
    int arr[n];
    printf("Enter array elements: ");
    for (int i = 0; i < n; i++)
        scanf("%d", &arr[i]);

    if (isMaxHeap(arr, n))
        printf("true\n");
    else
        printf("false\n");

    return 0;
}

File: DSA/LakshaKundra_590013237/Day_57_Question_2.py
def is_max_heap(arr):
    n = len(arr)
    for i in range(n // 2):
        if (2*i+1 < n and arr[i] < arr[2*i+1]) or (2*i+2 < n and arr[i] < arr[2*i+2]):
            return False
    return True

print(is_max_heap([90, 15, 10, 7, 12, 2]))
print(is_max_heap([9, 15, 10, 7, 12, 2]))
